### DISCUSSION TITLE
Typescript immer withImmer or anything to help, outdated code examples?

### DISCUSSION THREAD

**Discussion by AddictArts** [pmndrs/zustand#1359] [Q&A] [answered]
The below is no longer valid with the latest Zustand, does anyone have an updated version?

```
// https://docs.pmnd.rs/zustand/recipes#middleware
const withImmer =
  <TState extends State>(config: StateCreator<TState, (fn: (draft: Draft<TState>) => void) => void>): StateCreator<TState> =>
  (set, get, api) =>
    config((fn) => set(produce<TState>(fn)), get, api);
```

Thanks for the help.

**Comment by dai-shi** [comment, marked as answer]
check this: https://docs.pmnd.rs/zustand/integrations/updating-draft-states

**Reply by AddictArts** [reply]
> check this: https://docs.pmnd.rs/zustand/integrations/updating-draft-states

Thanks, but that doesn't work, wish it did. `immer<>` doesn't seem to be a thing in the latest.

Ugg, apologies. If you leave `create<>` as such you'll get the errors. Thanks this did end up helping.

**Reply by dai-shi** [reply]
How about `create<...>()(immer(...))`?

**Reply by AddictArts** [reply]
> How about `create<...>()(immer(...))`?

Probably would also work. It was my fault I forgot I had the type on the `create` call, oops.
